src/test/java/com/google/devtools/build/lib/bazel/repository/downloader/HttpStreamTest.java

```diff
@@ -105,6 +112,50 @@ public void smallDataWithValidChecksum_readsOk() throws Exception {
     }
   }
 
+  @Test
+  public void smallDataWithValidChecksum_timesOutInCreateRetriesOk() throws Exception {
+    InputStream inputStream = mock(ByteArrayInputStream.class);
+    InputStream realInputStream = new ByteArrayInputStream(data);
+
+    doAnswer((Answer<Integer>) invocation -> {
+      Object[] args = invocation.getArguments();
+      Object mock = invocation.getMock();
+
+      if (nRetries++ == 0) {
+        throw new SocketTimeoutException();
+      } else {
+        return realInputStream.read((byte[])args[0], (int)args[1], (int)args[2]);
+      }
+    }).when(inputStream).read(any(), anyInt(), anyInt());
+    when(reconnector.connect(any(), any())).thenReturn(connection);
+    when(connection.getInputStream()).thenReturn(inputStream);
+    when(connection.getHeaderField("Accept-Ranges")).thenReturn("bytes");
+    try (HttpStream stream = streamFactory.create(connection, AURL, GOOD_CHECKSUM, reconnector)) {
+      assertThat(toByteArray(stream)).isEqualTo(data);
+    }
+  }
+
+  @Test
+  public void smallDataWithValidChecksum_timesOutInCreateRepeatedly() throws Exception {
+    InputStream inputStream = mock(ByteArrayInputStream.class);
+
+    doAnswer((Answer<Integer>) invocation -> {
+      ++nRetries;
+      throw new SocketTimeoutException();
+    }).when(inputStream).read(any(), anyInt(), anyInt());
+    when(reconnector.connect(any(), any())).thenReturn(connection);
+    when(connection.getInputStream()).thenReturn(inputStream);
+    when(connection.getHeaderField("Accept-Ranges")).thenReturn("bytes");
+    thrown.expect(SocketTimeoutException.class);
+
+    try {
+      streamFactory.create(connection, AURL, GOOD_CHECKSUM, reconnector);
+    } catch (Exception e) {
+      assertThat(nRetries).isGreaterThan(3);  // RetryingInputStream.MAX_RESUMES
```

Comment:
might as well use RetryingInputStream.MAX_RESUMES itself in isGreaterThan()

Reply:
Making `MAX_RESUMES` part of `RetryingInputStream`'s public interface, or by using reflection?

Reminder: I'm not a Java developer. ;)

Reply:
yeah it's a bit of a bother, i assumed it was already public, we can just use 3